pkg/network/ebpf/c/http.h

```diff
@@ -210,16 +213,22 @@ static __always_inline int http_process(http_transaction_t *http_stack, skb_info
     http_method_t method = HTTP_METHOD_UNKNOWN;
     http_parse_data(buffer, &packet_type, &method);
 
-    http_transaction_t *http = http_fetch_state(http_stack, skb_info, packet_type);
-    if (http == NULL) {
+    http_transaction_t *http = http_fetch_state(http_stack, packet_type);
+    if (!http || http_seen_before(http, skb_info)) {
         return 0;
     }
 
     http_transaction_t *to_flush = http_should_flush_previous_state(http, packet_type);
+    if (to_flush) {
+        __builtin_memcpy(http, http_stack, sizeof(http_transaction_t));
+    }
```

Comment:
This is the actual fix: whenever `to_flush` is non-null  (which means there is a previous `http_transaction_t` associated to this connection that hasn't been flushed yet) we make sure to _fully_ override the eBPF map entry with the "new" data (instead of only overriding some fields)